File: Programmers/Inho/Level01/콜라츠 추측.cs
public class Solution {
    public int solution(double num) {
        int answer = 0;
        int count = 0;
        while(true){
            if(num == 1){
                return count;
            }
            count ++;
            if(count > 500 )
                return -1;
            if(num %2 == 0){
                num /=2;
            }
            else{
                num = num*3 +1;
            }
        }
        return answer;
    }
}

File: Programmers/Inho/C++/Level01/콜라츠 추측.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(int num) {
    int answer = 0;
    int count = 0;
    long num1= long(num);
    while(true)
    {
        if(num1 == 1)
            break;
        if(num1%2==0)
            num1/=2;
        else
        {
            num1*=3;
            num1++;
        }
        count++;
        if(count >=500)
            return -1;
        
    }
    return count;
}